src/lib.rs

```diff
@@ -350,3 +353,34 @@ pub fn getrandom_uninit(dest: &mut [MaybeUninit<u8>]) -> Result<&mut [u8], Error
     // since it returned `Ok`.
     Ok(unsafe { slice_assume_init_mut(dest) })
 }
+
+/// Generate a random value of type `T` implementing the [`zerocopy::FromBytes`] trait.
+///
+/// # Examples
+/// ```
+/// # fn main() -> Result<(), getrandom::Error> {
+/// let key: [u8; 16] = getrandom::value()?;
+/// let keys: [[u8; 16]; 64] = getrandom::value()?;
+/// let random_u32: u32 = getrandom::value()?;
+/// let random_u64s: [u64; 100] = getrandom::getrandom_value()?;
+/// # Ok(()) }
+/// ```
+#[cfg(feature = "zerocopy")]
+#[inline]
+pub fn value<T: zerocopy::FromBytes + Sized>() -> Result<T, Error> {
+    let mut value = MaybeUninit::<T>::uninit();
+    // SAFETY: it's safe to cast `&mut MaybeUninit<T>` to `&mut [MaybeUninit<u8>]`
+    // with slice length equal to `size_of::<T>()`. The compiler will ensure that
+    // `T` isn't too large.
+    unsafe {
+        let as_bytes_mut = core::slice::from_raw_parts_mut(
+            &mut value as *mut MaybeUninit<T> as *mut MaybeUninit<u8>,
```

Comment:
The second `as` at lesat should be `cast()`. If there's a way to replace `as` in the ref->pointer cast then that change should be made too.

Actually, instead of using this casting, isn't the purpose of `zerocopy::FromBytes` to provide an ergonomic way of doing this cast?

Reply:
I did not use `cast()` because of the crate MSRV, but I forgot that the `zerocopy` feature bumps MSRV, so it using `cast()` should be fine.

Reply:
>Actually, instead of using this casting, isn't the purpose of zerocopy::FromBytes to provide an ergonomic way of doing this cast?

I don't think that `zerocopy` currently provides such API, but it could be a worthwhile addition. Something likes this:
```rust
fn as_uninit_bytes_mut(val: &mut MaybeUninit<Self>) -> &mut [MaybeUninit<u8>] { ... }
```

cc @josephlr

Reply:
That sounds like [`MaybeUninit::as_bytes_mut`](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.as_bytes_mut). Though it's currently unstable.